Client/mods/deathmatch/logic/luadefs/CLuaBuildingDefs.cpp

```diff
@@ -48,14 +48,13 @@ CClientBuilding* CLuaBuildingDefs::CreateBuilding(lua_State* const luaVM, std::u
     if (!CClientBuildingManager::IsValidPosition(pos))
         throw std::invalid_argument("Position is outside of game world");
 
-    if (rot.has_value())
-        ConvertDegreesToRadians(rot.value());
-    else
-        rot.emplace(CVector(0, 0, 0));
+    CVector rot{rotX.value_or(0), rotY.value_or(0), rotZ.value_or(0)};
+    if (rot.fX != 0 || rot.fY != 0 || rot.fZ != 0)
+        ConvertDegreesToRadians(rot);
```

Comment:
There should be floating point literals (.0f). In addition to that `rot.fX != 0` can be replaced with `std::abs(rot.fX) < std::numeric_limits<float>::epsilon()` to make the comparison less strong.

Reply:
Done